shell/browser/native_window_mac.mm

```diff
@@ -1402,13 +1402,14 @@ void ReorderChildWindowAbove(NSWindow* child_window, NSWindow* other_window) {
     const bool no_rounded_corner = !HasStyleMask(NSWindowStyleMaskTitled);
     const int macos_version = base::mac::MacOSMajorVersion();
 
-    // Modal window corners are rounded on macOS >= 11 or higher if the user
-    // hasn't passed noRoundedCorners.
+    // If the window is modal, its corners are rounded on macOS >= 11 or higher
+    // unless the user has explicitly passed noRoundedCorners.
     bool should_round_modal =
-        !no_rounded_corner && (macos_version >= 11 ? true : !is_modal());
-    // Nonmodal window corners are rounded if they're frameless and the user
-    // hasn't passed noRoundedCorners.
-    bool should_round_nonmodal = !no_rounded_corner && !has_frame();
+        is_modal() ? (macos_version >= 11 && !no_rounded_corner) : false;
```

Comment:
_(nits)_

- This logic is better than the previous code but could be even simpler; that `: false` isn't needed

- Slightly more readable if the logic in this test & the next test both begin with `!no_rounded_corner`

```suggestion
        !no_rounded_corner && macos_version >= 11 && is_modal();
```